api/v1/submissions/PKPSubmissionController.php

```diff
@@ -819,6 +827,50 @@ public function delete(Request $illuminateRequest): JsonResponse
         return response()->json($submissionProps, Response::HTTP_OK);
     }
 
+    /**
+     * Change submission language
+     */
+    public function changeLocale(Request $illuminateRequest): JsonResponse
+    {
+        $publication = Repo::publication()->get((int) $illuminateRequest->route('publicationId'));
+
+        if (!$publication) {
+            return response()->json([
+                'error' => __('api.404.resourceNotFound'),
+            ], Response::HTTP_NOT_FOUND);
+        }
+
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+
+        if ($submission->getId() !== $publication->getData('submissionId')) {
+            return response()->json([
+                'error' => __('api.publications.403.submissionsDidNotMatch'),
+            ], Response::HTTP_FORBIDDEN);
+        }
+
+        $paramsSubmission = $this->convertStringsToSchema(PKPSchemaService::SCHEMA_SUBMISSION, $illuminateRequest->input());
+        $newLocale = $paramsSubmission['locale'] ?? null;
+
+        // Submission language can not be changed when there are more than one publication or a publication's status is published
+        if (!$newLocale || count($submission->getData('publications')) > 1 || $publication->getData('status') === PKPSubmission::STATUS_PUBLISHED) {
+            return response()->json(['error' => __('api.submission.403.cantChangeSubmissionLanguage')], Response::HTTP_FORBIDDEN);
+        }
+
+        // Convert a form field value to multilingual (if it is not) and merge rest values
+        collect(app()->get('schema')->getMultilingualProps(PKPSchemaService::SCHEMA_PUBLICATION))
+            ->each(fn (string $p) => $illuminateRequest->whenHas($p, fn ($v) => $illuminateRequest->merge([$p => array_merge($publication->getData($p) ?? [], is_array($v) ? $v : [$newLocale => $v])])));
```

Comment:
I don't have any immediate solution or recommendation, but just wanted to flag it here while reviewing the rest: I'm a little unsure about directly modifying the incoming request in order to complete this work, especially with the approach of passing on the request to two further API endpoints. I agree that this is a good solution to not having to rewrite code that already exist, but I'm not sure if this is a pattern we'd want to adopt/encourage. I'll do a bit of thinking and come back to it.

Reply:
@ewhanson Just a thought on this and also not sure how its applicable to our existing code base. But one strategy to take is to keep the API handlers very slim&lightweight, with just API specific things like validation. And than having business logic in other application layers, where that logic can be invoked from anywhere and its easier to reuse.

Reply:
Thanks @jardakotesovec. I like that idea in general and I think this use case would be a good candidate for testing that out. I think that kind of change would be outside the scope of this issue so in the interest in keeping things moving, I would recommend keeping this as-is for this issue, and creating a new issue to look into implementing that with this as a first test case.